File: java/DataFrame/src/main/java/coyote/dataframe/U16Type.java
package coyote.dataframe;

import coyote.util.ByteUtil;

/** (5) Type code representing an unsigned, 16-bit value in the range of 0 to 65,535 */
public class U16Type implements FieldType
{
  private static final int _size = 2;

  private final static String _name = "U16";




  public boolean checkType( Object obj )
  {
    return obj instanceof Short;
  }




  public Object decode( byte[] value )
  {
    return new Integer( ByteUtil.retrieveUnsignedShort( value, 0 ) );
  }




  public byte[] encode( Object obj )
  {
    return ByteUtil.renderUnsignedShort( ( (Short)obj ).shortValue() );
  }




  public String getTypeName()
  {
    return _name;
  }




  public boolean isNumeric()
  {
    return true;
  }




  public int getSize()
  {
    return _size;
  }

}

File: csharp/DataFrame/Coyote/DataFrame/U16Type.cs
#region license
// Copyright (c) 2006 Stephan D. Cote' - All rights reserved.
// 
// This program and the accompanying materials are made available under the 
// terms of the MIT License which accompanies this distribution, and is 
// available at http://creativecommons.org/licenses/MIT/
// *
// Contributors:
//   Stephan D. Cote 
//      - Initial API and implementation
#endregion

namespace Coyote.DataFrame {



    /// <summary>This is a numeric type equivalent to the <code>ushort</code> type, representing an unsigned, 16-bit value in the range of 0 to 65,535 </summary>
    public class U16Type : FieldType {
        private const int _size = 2;

        private const string _name = "U16";


        public virtual string TypeName {
            get {
                return _name;
            }
        }




        public virtual bool IsNumeric {
            get {
                return true;
            }
        }




        public virtual int Size {
            get {
                return _size;
            }
        }




        public virtual bool CheckType( object obj ) {
            return (obj is ushort);
        }




        public virtual object Decode( byte[] value ) {
            return ByteUtil.RetrieveUnsignedShort( value, 0 );
        }




        public virtual byte[] Encode( object obj ) {
            return ByteUtil.RenderUnsignedShort( (ushort)obj );
        }




        public virtual string StringValue( byte[] val ) {
            if ( val == null ) {
                return "";
            } else {
                object obj = Decode( val );
                if ( obj != null )
                    return obj.ToString();
                else
                    return "";
            }
        }

    }

}